src/main/java/com/nutomic/syncthingandroid/activities/LogActivity.java

```diff
@@ -0,0 +1,168 @@
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        if (savedInstanceState != null) {
+            mSyncthingLog = savedInstanceState.getBoolean("syncthingLog");
+        }
+
+        mLog = (TextView) findViewById(R.id.log);
+        mScrollView = (ScrollView) findViewById(R.id.scroller);
+
+        updateLog();
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putBoolean("syncthingLog", mSyncthingLog);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.log_list, menu);
+
+        // Add the share button
+        MenuItem shareItem = menu.findItem(R.id.menu_share);
+        if (mSyncthingLog) {
+            shareItem.setTitle(R.string.log_android_title);
+        } else {
+            shareItem.setTitle(R.string.log_syncthing_title);
+        }
+        ShareActionProvider actionProvider = (ShareActionProvider) MenuItemCompat.getActionProvider(shareItem);
+        mShareIntent = new Intent();
+        mShareIntent.setAction(Intent.ACTION_SEND);
+        mShareIntent.setType("text/plain");
+        mShareIntent.putExtra(android.content.Intent.EXTRA_TEXT, mLog.getText());
+        actionProvider.setShareIntent(mShareIntent);
+
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.switch_logs:
+                mSyncthingLog = !mSyncthingLog;
+                if (mSyncthingLog) {
+                    item.setTitle(R.string.log_android_title);
+                } else {
+                    item.setTitle(R.string.log_syncthing_title);
+                }
+                updateLog();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    private void scrollToBottom() {
+        mScrollView.post(new Runnable() {
+            @Override
+            public void run() {
+                mScrollView.scrollTo(0, mLog.getBottom());
+            }
+        });
+    }
+
+    private void updateLog() {
+        if (mFetchLogTask != null)
+            mFetchLogTask.cancel(true);
+        mLog.setText("Retrieving logs...");
+        mFetchLogTask = new AsyncTask<Void, Void, String>() {
+            @Override
+            protected String doInBackground(Void... params) {
+                return getLog(mSyncthingLog);
+            }
+            @Override
+            protected void onPostExecute(String log) {
+                mLog.setText(log);
+                if (mShareIntent != null)
+                    mShareIntent.putExtra(android.content.Intent.EXTRA_TEXT, log);
+                scrollToBottom();
+            }
+        }.execute();
+    }
+
+    private String getLog(final boolean syncthingLog) {
```

Comment:
We're optimizing for humans, not for the compiler. I'm pretty sure the compiler can make the same code from both, but one is much easier to read.

And lambdas are different because they have little to no boilerplate code.